app/code/community/Bolt/Boltpay/Block/Checkout/Boltpay.php

```diff
@@ -228,7 +259,7 @@ public function buildBoltCheckoutJavascript($checkoutType, $immutableQuoteId, $h
         /* @var Bolt_Boltpay_Helper_Api $boltHelper */
         $boltHelper = Mage::helper('boltpay');
 
-        $jsonCart = json_encode($cartData);
+        $jsonCart = (!is_string($cartData)) ? json_encode($cartData) : $cartData;
```

Comment:
`is_string($cartData)) ? $cartData : json_encode($cartData)`
(avoid negation for readability)